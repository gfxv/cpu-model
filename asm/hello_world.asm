
org 20
hello_string:
    word "Hello World"

_start:
    ld &hello_string

    ; Preparation
    st #4   ; Pointer to string
    ld $4    ; Load length (acts like counter)
    st #5   ; Store counter with address 5

    ; [4] - ptr to string
    ; [5] - total length 

    loop:
    ; inc ptr
    ld 4   ; load current pointer
    inc    ; increment pointer
    st #4  ; save

    ; out char
    ld $4
    st #1  ; [1] - i/o address

    ; dec counter
    ld 5   ; load counter
    dec    ; decrement
    st #5  ; savve

    ; check length
    jz stop
    jmp loop

    stop:
    hlt





