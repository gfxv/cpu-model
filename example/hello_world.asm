
org 20

hello_string:
    word "Hello, World!"
pointer:
    int 0
counter:
    int 0

_start:
    ld &hello_string   ; load pointer
    st &pointer

    ld hello_string    ; Load length (acts like counter)
    st &counter        ; Store counter with address 5

    loop:
        ; inc ptr
        ld pointer   ; load current pointer
        inc          ; increment pointer
        st &pointer  ; save

        ; output symbol
        ld $pointer
        st #2

        ; dec counter
        ld counter   ; load counter
        dec          ; decrement
        st &counter  ; savve

        ; check length
        jz stop
        jmp loop

    stop:
    hlt
