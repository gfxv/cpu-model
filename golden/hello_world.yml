source: |-
  org 20

  hello_string:
      word "Hello, World!"
  pointer:
      int 0
  counter:
      int 0

  _start:
      ld &hello_string
      st &pointer

      ld hello_string
      st &counter

      loop:
          ld pointer
          inc
          st &pointer

          ld $pointer
          st #1

          ld counter
          dec
          st &counter

          jz stop
          jmp loop

      stop:
      hlt
in: |-
  0
out_code: |-
  [
    {
      "index": 0,
      "opcode": "jmp",
      "arg": "37",
      "arg_type": "default"
    },
    {
      "index": 21,
      "opcode": "nop",
      "arg": 13,
      "arg_type": "data"
    },
    {
      "index": 22,
      "opcode": "nop",
      "arg": 72,
      "arg_type": "data"
    },
    {
      "index": 23,
      "opcode": "nop",
      "arg": 101,
      "arg_type": "data"
    },
    {
      "index": 24,
      "opcode": "nop",
      "arg": 108,
      "arg_type": "data"
    },
    {
      "index": 25,
      "opcode": "nop",
      "arg": 108,
      "arg_type": "data"
    },
    {
      "index": 26,
      "opcode": "nop",
      "arg": 111,
      "arg_type": "data"
    },
    {
      "index": 27,
      "opcode": "nop",
      "arg": 44,
      "arg_type": "data"
    },
    {
      "index": 28,
      "opcode": "nop",
      "arg": 32,
      "arg_type": "data"
    },
    {
      "index": 29,
      "opcode": "nop",
      "arg": 87,
      "arg_type": "data"
    },
    {
      "index": 30,
      "opcode": "nop",
      "arg": 111,
      "arg_type": "data"
    },
    {
      "index": 31,
      "opcode": "nop",
      "arg": 114,
      "arg_type": "data"
    },
    {
      "index": 32,
      "opcode": "nop",
      "arg": 108,
      "arg_type": "data"
    },
    {
      "index": 33,
      "opcode": "nop",
      "arg": 100,
      "arg_type": "data"
    },
    {
      "index": 34,
      "opcode": "nop",
      "arg": 33,
      "arg_type": "data"
    },
    {
      "index": 35,
      "opcode": "nop",
      "arg": 0,
      "arg_type": "data"
    },
    {
      "index": 36,
      "opcode": "nop",
      "arg": 0,
      "arg_type": "data"
    },
    {
      "index": 37,
      "opcode": "ld",
      "arg": 21,
      "arg_type": "raw"
    },
    {
      "index": 38,
      "opcode": "st",
      "arg": 35,
      "arg_type": "raw"
    },
    {
      "index": 39,
      "opcode": "ld",
      "arg": 21,
      "arg_type": "default"
    },
    {
      "index": 40,
      "opcode": "st",
      "arg": 36,
      "arg_type": "raw"
    },
    {
      "index": 41,
      "opcode": "ld",
      "arg": 35,
      "arg_type": "default"
    },
    {
      "index": 42,
      "opcode": "inc",
      "arg_type": "none",
      "arg": 0
    },
    {
      "index": 43,
      "opcode": "st",
      "arg": 35,
      "arg_type": "raw"
    },
    {
      "index": 44,
      "opcode": "ld",
      "arg": 35,
      "arg_type": "ptr"
    },
    {
      "index": 45,
      "opcode": "st",
      "arg": "1",
      "arg_type": "raw"
    },
    {
      "index": 46,
      "opcode": "ld",
      "arg": 36,
      "arg_type": "default"
    },
    {
      "index": 47,
      "opcode": "dec",
      "arg_type": "none",
      "arg": 0
    },
    {
      "index": 48,
      "opcode": "st",
      "arg": 36,
      "arg_type": "raw"
    },
    {
      "index": 49,
      "opcode": "jz",
      "arg": 51,
      "arg_type": "default"
    },
    {
      "index": 50,
      "opcode": "jmp",
      "arg": 41,
      "arg_type": "default"
    },
    {
      "index": 51,
      "opcode": "hlt",
      "arg_type": "none",
      "arg": 0
    }
  ]
out: |
  Instructions: 135
  Ticks: 770
  === OUT ===
  --- bytes ---
  72 101 108 108 111 44 32 87 111 114 108 100 33
  --- string ---
  Hello, World!
log: |
  DEBUG    root:control_unit.py:106 TICK: 4    | jmp  | ARG: 37  | TYPE: default | PC: 37  | ACC: 0   | AR: 0   | N: False  | Z: False
  DEBUG    root:control_unit.py:106 TICK: 9    | ld   | ARG: 21  | TYPE: raw     | PC: 38  | ACC: 21  | AR: 37  | N: False  | Z: False
  DEBUG    root:control_unit.py:106 TICK: 16   | st   | ARG: 35  | TYPE: raw     | PC: 39  | ACC: 21  | AR: 35  | N: False  | Z: False
  DEBUG    root:control_unit.py:106 TICK: 22   | ld   | ARG: 21  | TYPE: default | PC: 40  | ACC: 13  | AR: 21  | N: False  | Z: False
  DEBUG    root:control_unit.py:106 TICK: 29   | st   | ARG: 36  | TYPE: raw     | PC: 41  | ACC: 13  | AR: 36  | N: False  | Z: False
  DEBUG    root:control_unit.py:106 TICK: 35   | ld   | ARG: 35  | TYPE: default | PC: 42  | ACC: 21  | AR: 35  | N: False  | Z: False
  DEBUG    root:control_unit.py:106 TICK: 39   | inc  | ARG: -   | TYPE: none    | PC: 43  | ACC: 22  | AR: 42  | N: False  | Z: False
  DEBUG    root:control_unit.py:106 TICK: 46   | st   | ARG: 35  | TYPE: raw     | PC: 44  | ACC: 22  | AR: 35  | N: False  | Z: False
  DEBUG    root:control_unit.py:106 TICK: 54   | ld   | ARG: 35  | TYPE: ptr     | PC: 45  | ACC: 72  | AR: 22  | N: False  | Z: False
  DEBUG    root:control_unit.py:106 TICK: 61   | st   | ARG: 1   | TYPE: raw     | PC: 46  | ACC: 72  | AR: 1   | N: False  | Z: False
  DEBUG    root:control_unit.py:106 TICK: 67   | ld   | ARG: 36  | TYPE: default | PC: 47  | ACC: 13  | AR: 36  | N: False  | Z: False
  DEBUG    root:control_unit.py:106 TICK: 71   | dec  | ARG: -   | TYPE: none    | PC: 48  | ACC: 12  | AR: 47  | N: False  | Z: False
  DEBUG    root:control_unit.py:106 TICK: 78   | st   | ARG: 36  | TYPE: raw     | PC: 49  | ACC: 12  | AR: 36  | N: False  | Z: False
  DEBUG    root:control_unit.py:106 TICK: 82   | jz   | ARG: 51  | TYPE: default | PC: 50  | ACC: 12  | AR: 49  | N: False  | Z: False
  DEBUG    root:control_unit.py:106 TICK: 86   | jmp  | ARG: 41  | TYPE: default | PC: 41  | ACC: 12  | AR: 50  | N: False  | Z: False
  DEBUG    root:control_unit.py:106 TICK: 92   | ld   | ARG: 35  | TYPE: default | PC: 42  | ACC: 22  | AR: 35  | N: False  | Z: False
  DEBUG    root:control_unit.py:106 TICK: 96   | inc  | ARG: -   | TYPE: none    | PC: 43  | ACC: 23  | AR: 42  | N: False  | Z: False
  DEBUG    root:control_unit.py:106 TICK: 103  | st   | ARG: 35  | TYPE: raw     | PC: 44  | ACC: 23  | AR: 35  | N: False  | Z: False
  DEBUG    root:control_unit.py:106 TICK: 111  | ld   | ARG: 35  | TYPE: ptr     | PC: 45  | ACC: 101 | AR: 23  | N: False  | Z: False
  DEBUG    root:control_unit.py:106 TICK: 118  | st   | ARG: 1   | TYPE: raw     | PC: 46  | ACC: 101 | AR: 1   | N: False  | Z: False
  DEBUG    root:control_unit.py:106 TICK: 124  | ld   | ARG: 36  | TYPE: default | PC: 47  | ACC: 12  | AR: 36  | N: False  | Z: False
  DEBUG    root:control_unit.py:106 TICK: 128  | dec  | ARG: -   | TYPE: none    | PC: 48  | ACC: 11  | AR: 47  | N: False  | Z: False
  DEBUG    root:control_unit.py:106 TICK: 135  | st   | ARG: 36  | TYPE: raw     | PC: 49  | ACC: 11  | AR: 36  | N: False  | Z: False
  DEBUG    root:control_unit.py:106 TICK: 139  | jz   | ARG: 51  | TYPE: default | PC: 50  | ACC: 11  | AR: 49  | N: False  | Z: False
  DEBUG    root:control_unit.py:106 TICK: 143  | jmp  | ARG: 41  | TYPE: default | PC: 41  | ACC: 11  | AR: 50  | N: False  | Z: False
  DEBUG    root:control_unit.py:106 TICK: 149  | ld   | ARG: 35  | TYPE: default | PC: 42  | ACC: 23  | AR: 35  | N: False  | Z: False
  DEBUG    root:control_unit.py:106 TICK: 153  | inc  | ARG: -   | TYPE: none    | PC: 43  | ACC: 24  | AR: 42  | N: False  | Z: False
  DEBUG    root:control_unit.py:106 TICK: 160  | st   | ARG: 35  | TYPE: raw     | PC: 44  | ACC: 24  | AR: 35  | N: False  | Z: False
  DEBUG    root:control_unit.py:106 TICK: 168  | ld   | ARG: 35  | TYPE: ptr     | PC: 45  | ACC: 108 | AR: 24  | N: False  | Z: False
  DEBUG    root:control_unit.py:106 TICK: 175  | st   | ARG: 1   | TYPE: raw     | PC: 46  | ACC: 108 | AR: 1   | N: False  | Z: False
  DEBUG    root:control_unit.py:106 TICK: 181  | ld   | ARG: 36  | TYPE: default | PC: 47  | ACC: 11  | AR: 36  | N: False  | Z: False
  DEBUG    root:control_unit.py:106 TICK: 185  | dec  | ARG: -   | TYPE: none    | PC: 48  | ACC: 10  | AR: 47  | N: False  | Z: False
  DEBUG    root:control_unit.py:106 TICK: 192  | st   | ARG: 36  | TYPE: raw     | PC: 49  | ACC: 10  | AR: 36  | N: False  | Z: False
  DEBUG    root:control_unit.py:106 TICK: 196  | jz   | ARG: 51  | TYPE: default | PC: 50  | ACC: 10  | AR: 49  | N: False  | Z: False
  DEBUG    root:control_unit.py:106 TICK: 200  | jmp  | ARG: 41  | TYPE: default | PC: 41  | ACC: 10  | AR: 50  | N: False  | Z: False
  DEBUG    root:control_unit.py:106 TICK: 206  | ld   | ARG: 35  | TYPE: default | PC: 42  | ACC: 24  | AR: 35  | N: False  | Z: False
  DEBUG    root:control_unit.py:106 TICK: 210  | inc  | ARG: -   | TYPE: none    | PC: 43  | ACC: 25  | AR: 42  | N: False  | Z: False
  DEBUG    root:control_unit.py:106 TICK: 217  | st   | ARG: 35  | TYPE: raw     | PC: 44  | ACC: 25  | AR: 35  | N: False  | Z: False
  DEBUG    root:control_unit.py:106 TICK: 225  | ld   | ARG: 35  | TYPE: ptr     | PC: 45  | ACC: 108 | AR: 25  | N: False  | Z: False
  DEBUG    root:control_unit.py:106 TICK: 232  | st   | ARG: 1   | TYPE: raw     | PC: 46  | ACC: 108 | AR: 1   | N: False  | Z: False
  DEBUG    root:control_unit.py:106 TICK: 238  | ld   | ARG: 36  | TYPE: default | PC: 47  | ACC: 10  | AR: 36  | N: False  | Z: False
  DEBUG    root:control_unit.py:106 TICK: 242  | dec  | ARG: -   | TYPE: none    | PC: 48  | ACC: 9   | AR: 47  | N: False  | Z: False
  DEBUG    root:control_unit.py:106 TICK: 249  | st   | ARG: 36  | TYPE: raw     | PC: 49  | ACC: 9   | AR: 36  | N: False  | Z: False
  DEBUG    root:control_unit.py:106 TICK: 253  | jz   | ARG: 51  | TYPE: default | PC: 50  | ACC: 9   | AR: 49  | N: False  | Z: False
  DEBUG    root:control_unit.py:106 TICK: 257  | jmp  | ARG: 41  | TYPE: default | PC: 41  | ACC: 9   | AR: 50  | N: False  | Z: False
  DEBUG    root:control_unit.py:106 TICK: 263  | ld   | ARG: 35  | TYPE: default | PC: 42  | ACC: 25  | AR: 35  | N: False  | Z: False
  DEBUG    root:control_unit.py:106 TICK: 267  | inc  | ARG: -   | TYPE: none    | PC: 43  | ACC: 26  | AR: 42  | N: False  | Z: False
  DEBUG    root:control_unit.py:106 TICK: 274  | st   | ARG: 35  | TYPE: raw     | PC: 44  | ACC: 26  | AR: 35  | N: False  | Z: False
  DEBUG    root:control_unit.py:106 TICK: 282  | ld   | ARG: 35  | TYPE: ptr     | PC: 45  | ACC: 111 | AR: 26  | N: False  | Z: False
  DEBUG    root:control_unit.py:106 TICK: 289  | st   | ARG: 1   | TYPE: raw     | PC: 46  | ACC: 111 | AR: 1   | N: False  | Z: False
  DEBUG    root:control_unit.py:106 TICK: 295  | ld   | ARG: 36  | TYPE: default | PC: 47  | ACC: 9   | AR: 36  | N: False  | Z: False
  DEBUG    root:control_unit.py:106 TICK: 299  | dec  | ARG: -   | TYPE: none    | PC: 48  | ACC: 8   | AR: 47  | N: False  | Z: False
  DEBUG    root:control_unit.py:106 TICK: 306  | st   | ARG: 36  | TYPE: raw     | PC: 49  | ACC: 8   | AR: 36  | N: False  | Z: False
  DEBUG    root:control_unit.py:106 TICK: 310  | jz   | ARG: 51  | TYPE: default | PC: 50  | ACC: 8   | AR: 49  | N: False  | Z: False
  DEBUG    root:control_unit.py:106 TICK: 314  | jmp  | ARG: 41  | TYPE: default | PC: 41  | ACC: 8   | AR: 50  | N: False  | Z: False
  DEBUG    root:control_unit.py:106 TICK: 320  | ld   | ARG: 35  | TYPE: default | PC: 42  | ACC: 26  | AR: 35  | N: False  | Z: False
  DEBUG    root:control_unit.py:106 TICK: 324  | inc  | ARG: -   | TYPE: none    | PC: 43  | ACC: 27  | AR: 42  | N: False  | Z: False
  DEBUG    root:control_unit.py:106 TICK: 331  | st   | ARG: 35  | TYPE: raw     | PC: 44  | ACC: 27  | AR: 35  | N: False  | Z: False
  DEBUG    root:control_unit.py:106 TICK: 339  | ld   | ARG: 35  | TYPE: ptr     | PC: 45  | ACC: 44  | AR: 27  | N: False  | Z: False
  DEBUG    root:control_unit.py:106 TICK: 346  | st   | ARG: 1   | TYPE: raw     | PC: 46  | ACC: 44  | AR: 1   | N: False  | Z: False
  DEBUG    root:control_unit.py:106 TICK: 352  | ld   | ARG: 36  | TYPE: default | PC: 47  | ACC: 8   | AR: 36  | N: False  | Z: False
  DEBUG    root:control_unit.py:106 TICK: 356  | dec  | ARG: -   | TYPE: none    | PC: 48  | ACC: 7   | AR: 47  | N: False  | Z: False
  DEBUG    root:control_unit.py:106 TICK: 363  | st   | ARG: 36  | TYPE: raw     | PC: 49  | ACC: 7   | AR: 36  | N: False  | Z: False
  DEBUG    root:control_unit.py:106 TICK: 367  | jz   | ARG: 51  | TYPE: default | PC: 50  | ACC: 7   | AR: 49  | N: False  | Z: False
  DEBUG    root:control_unit.py:106 TICK: 371  | jmp  | ARG: 41  | TYPE: default | PC: 41  | ACC: 7   | AR: 50  | N: False  | Z: False
  DEBUG    root:control_unit.py:106 TICK: 377  | ld   | ARG: 35  | TYPE: default | PC: 42  | ACC: 27  | AR: 35  | N: False  | Z: False
  DEBUG    root:control_unit.py:106 TICK: 381  | inc  | ARG: -   | TYPE: none    | PC: 43  | ACC: 28  | AR: 42  | N: False  | Z: False
  DEBUG    root:control_unit.py:106 TICK: 388  | st   | ARG: 35  | TYPE: raw     | PC: 44  | ACC: 28  | AR: 35  | N: False  | Z: False
  DEBUG    root:control_unit.py:106 TICK: 396  | ld   | ARG: 35  | TYPE: ptr     | PC: 45  | ACC: 32  | AR: 28  | N: False  | Z: False
  DEBUG    root:control_unit.py:106 TICK: 403  | st   | ARG: 1   | TYPE: raw     | PC: 46  | ACC: 32  | AR: 1   | N: False  | Z: False
  DEBUG    root:control_unit.py:106 TICK: 409  | ld   | ARG: 36  | TYPE: default | PC: 47  | ACC: 7   | AR: 36  | N: False  | Z: False
  DEBUG    root:control_unit.py:106 TICK: 413  | dec  | ARG: -   | TYPE: none    | PC: 48  | ACC: 6   | AR: 47  | N: False  | Z: False
  DEBUG    root:control_unit.py:106 TICK: 420  | st   | ARG: 36  | TYPE: raw     | PC: 49  | ACC: 6   | AR: 36  | N: False  | Z: False
  DEBUG    root:control_unit.py:106 TICK: 424  | jz   | ARG: 51  | TYPE: default | PC: 50  | ACC: 6   | AR: 49  | N: False  | Z: False
  DEBUG    root:control_unit.py:106 TICK: 428  | jmp  | ARG: 41  | TYPE: default | PC: 41  | ACC: 6   | AR: 50  | N: False  | Z: False
  DEBUG    root:control_unit.py:106 TICK: 434  | ld   | ARG: 35  | TYPE: default | PC: 42  | ACC: 28  | AR: 35  | N: False  | Z: False
  DEBUG    root:control_unit.py:106 TICK: 438  | inc  | ARG: -   | TYPE: none    | PC: 43  | ACC: 29  | AR: 42  | N: False  | Z: False
  DEBUG    root:control_unit.py:106 TICK: 445  | st   | ARG: 35  | TYPE: raw     | PC: 44  | ACC: 29  | AR: 35  | N: False  | Z: False
  DEBUG    root:control_unit.py:106 TICK: 453  | ld   | ARG: 35  | TYPE: ptr     | PC: 45  | ACC: 87  | AR: 29  | N: False  | Z: False
  DEBUG    root:control_unit.py:106 TICK: 460  | st   | ARG: 1   | TYPE: raw     | PC: 46  | ACC: 87  | AR: 1   | N: False  | Z: False
  DEBUG    root:control_unit.py:106 TICK: 466  | ld   | ARG: 36  | TYPE: default | PC: 47  | ACC: 6   | AR: 36  | N: False  | Z: False
  DEBUG    root:control_unit.py:106 TICK: 470  | dec  | ARG: -   | TYPE: none    | PC: 48  | ACC: 5   | AR: 47  | N: False  | Z: False
  DEBUG    root:control_unit.py:106 TICK: 477  | st   | ARG: 36  | TYPE: raw     | PC: 49  | ACC: 5   | AR: 36  | N: False  | Z: False
  DEBUG    root:control_unit.py:106 TICK: 481  | jz   | ARG: 51  | TYPE: default | PC: 50  | ACC: 5   | AR: 49  | N: False  | Z: False
  DEBUG    root:control_unit.py:106 TICK: 485  | jmp  | ARG: 41  | TYPE: default | PC: 41  | ACC: 5   | AR: 50  | N: False  | Z: False
  DEBUG    root:control_unit.py:106 TICK: 491  | ld   | ARG: 35  | TYPE: default | PC: 42  | ACC: 29  | AR: 35  | N: False  | Z: False
  DEBUG    root:control_unit.py:106 TICK: 495  | inc  | ARG: -   | TYPE: none    | PC: 43  | ACC: 30  | AR: 42  | N: False  | Z: False
  DEBUG    root:control_unit.py:106 TICK: 502  | st   | ARG: 35  | TYPE: raw     | PC: 44  | ACC: 30  | AR: 35  | N: False  | Z: False
  DEBUG    root:control_unit.py:106 TICK: 510  | ld   | ARG: 35  | TYPE: ptr     | PC: 45  | ACC: 111 | AR: 30  | N: False  | Z: False
  DEBUG    root:control_unit.py:106 TICK: 517  | st   | ARG: 1   | TYPE: raw     | PC: 46  | ACC: 111 | AR: 1   | N: False  | Z: False
  DEBUG    root:control_unit.py:106 TICK: 523  | ld   | ARG: 36  | TYPE: default | PC: 47  | ACC: 5   | AR: 36  | N: False  | Z: False
  DEBUG    root:control_unit.py:106 TICK: 527  | dec  | ARG: -   | TYPE: none    | PC: 48  | ACC: 4   | AR: 47  | N: False  | Z: False
  DEBUG    root:control_unit.py:106 TICK: 534  | st   | ARG: 36  | TYPE: raw     | PC: 49  | ACC: 4   | AR: 36  | N: False  | Z: False
  DEBUG    root:control_unit.py:106 TICK: 538  | jz   | ARG: 51  | TYPE: default | PC: 50  | ACC: 4   | AR: 49  | N: False  | Z: False
  DEBUG    root:control_unit.py:106 TICK: 542  | jmp  | ARG: 41  | TYPE: default | PC: 41  | ACC: 4   | AR: 50  | N: False  | Z: False
  DEBUG    root:control_unit.py:106 TICK: 548  | ld   | ARG: 35  | TYPE: default | PC: 42  | ACC: 30  | AR: 35  | N: False  | Z: False
  DEBUG    root:control_unit.py:106 TICK: 552  | inc  | ARG: -   | TYPE: none    | PC: 43  | ACC: 31  | AR: 42  | N: False  | Z: False
  DEBUG    root:control_unit.py:106 TICK: 559  | st   | ARG: 35  | TYPE: raw     | PC: 44  | ACC: 31  | AR: 35  | N: False  | Z: False
  DEBUG    root:control_unit.py:106 TICK: 567  | ld   | ARG: 35  | TYPE: ptr     | PC: 45  | ACC: 114 | AR: 31  | N: False  | Z: False
  DEBUG    root:control_unit.py:106 TICK: 574  | st   | ARG: 1   | TYPE: raw     | PC: 46  | ACC: 114 | AR: 1   | N: False  | Z: False
  DEBUG    root:control_unit.py:106 TICK: 580  | ld   | ARG: 36  | TYPE: default | PC: 47  | ACC: 4   | AR: 36  | N: False  | Z: False
  DEBUG    root:control_unit.py:106 TICK: 584  | dec  | ARG: -   | TYPE: none    | PC: 48  | ACC: 3   | AR: 47  | N: False  | Z: False
  DEBUG    root:control_unit.py:106 TICK: 591  | st   | ARG: 36  | TYPE: raw     | PC: 49  | ACC: 3   | AR: 36  | N: False  | Z: False
  DEBUG    root:control_unit.py:106 TICK: 595  | jz   | ARG: 51  | TYPE: default | PC: 50  | ACC: 3   | AR: 49  | N: False  | Z: False
  DEBUG    root:control_unit.py:106 TICK: 599  | jmp  | ARG: 41  | TYPE: default | PC: 41  | ACC: 3   | AR: 50  | N: False  | Z: False
  DEBUG    root:control_unit.py:106 TICK: 605  | ld   | ARG: 35  | TYPE: default | PC: 42  | ACC: 31  | AR: 35  | N: False  | Z: False
  DEBUG    root:control_unit.py:106 TICK: 609  | inc  | ARG: -   | TYPE: none    | PC: 43  | ACC: 32  | AR: 42  | N: False  | Z: False
  DEBUG    root:control_unit.py:106 TICK: 616  | st   | ARG: 35  | TYPE: raw     | PC: 44  | ACC: 32  | AR: 35  | N: False  | Z: False
  DEBUG    root:control_unit.py:106 TICK: 624  | ld   | ARG: 35  | TYPE: ptr     | PC: 45  | ACC: 108 | AR: 32  | N: False  | Z: False
  DEBUG    root:control_unit.py:106 TICK: 631  | st   | ARG: 1   | TYPE: raw     | PC: 46  | ACC: 108 | AR: 1   | N: False  | Z: False
  DEBUG    root:control_unit.py:106 TICK: 637  | ld   | ARG: 36  | TYPE: default | PC: 47  | ACC: 3   | AR: 36  | N: False  | Z: False
  DEBUG    root:control_unit.py:106 TICK: 641  | dec  | ARG: -   | TYPE: none    | PC: 48  | ACC: 2   | AR: 47  | N: False  | Z: False
  DEBUG    root:control_unit.py:106 TICK: 648  | st   | ARG: 36  | TYPE: raw     | PC: 49  | ACC: 2   | AR: 36  | N: False  | Z: False
  DEBUG    root:control_unit.py:106 TICK: 652  | jz   | ARG: 51  | TYPE: default | PC: 50  | ACC: 2   | AR: 49  | N: False  | Z: False
  DEBUG    root:control_unit.py:106 TICK: 656  | jmp  | ARG: 41  | TYPE: default | PC: 41  | ACC: 2   | AR: 50  | N: False  | Z: False
  DEBUG    root:control_unit.py:106 TICK: 662  | ld   | ARG: 35  | TYPE: default | PC: 42  | ACC: 32  | AR: 35  | N: False  | Z: False
  DEBUG    root:control_unit.py:106 TICK: 666  | inc  | ARG: -   | TYPE: none    | PC: 43  | ACC: 33  | AR: 42  | N: False  | Z: False
  DEBUG    root:control_unit.py:106 TICK: 673  | st   | ARG: 35  | TYPE: raw     | PC: 44  | ACC: 33  | AR: 35  | N: False  | Z: False
  DEBUG    root:control_unit.py:106 TICK: 681  | ld   | ARG: 35  | TYPE: ptr     | PC: 45  | ACC: 100 | AR: 33  | N: False  | Z: False
  DEBUG    root:control_unit.py:106 TICK: 688  | st   | ARG: 1   | TYPE: raw     | PC: 46  | ACC: 100 | AR: 1   | N: False  | Z: False
  DEBUG    root:control_unit.py:106 TICK: 694  | ld   | ARG: 36  | TYPE: default | PC: 47  | ACC: 2   | AR: 36  | N: False  | Z: False
  DEBUG    root:control_unit.py:106 TICK: 698  | dec  | ARG: -   | TYPE: none    | PC: 48  | ACC: 1   | AR: 47  | N: False  | Z: False
  DEBUG    root:control_unit.py:106 TICK: 705  | st   | ARG: 36  | TYPE: raw     | PC: 49  | ACC: 1   | AR: 36  | N: False  | Z: False
  DEBUG    root:control_unit.py:106 TICK: 709  | jz   | ARG: 51  | TYPE: default | PC: 50  | ACC: 1   | AR: 49  | N: False  | Z: False
  DEBUG    root:control_unit.py:106 TICK: 713  | jmp  | ARG: 41  | TYPE: default | PC: 41  | ACC: 1   | AR: 50  | N: False  | Z: False
  DEBUG    root:control_unit.py:106 TICK: 719  | ld   | ARG: 35  | TYPE: default | PC: 42  | ACC: 33  | AR: 35  | N: False  | Z: False
  DEBUG    root:control_unit.py:106 TICK: 723  | inc  | ARG: -   | TYPE: none    | PC: 43  | ACC: 34  | AR: 42  | N: False  | Z: False
  DEBUG    root:control_unit.py:106 TICK: 730  | st   | ARG: 35  | TYPE: raw     | PC: 44  | ACC: 34  | AR: 35  | N: False  | Z: False
  DEBUG    root:control_unit.py:106 TICK: 738  | ld   | ARG: 35  | TYPE: ptr     | PC: 45  | ACC: 33  | AR: 34  | N: False  | Z: False
  DEBUG    root:control_unit.py:106 TICK: 745  | st   | ARG: 1   | TYPE: raw     | PC: 46  | ACC: 33  | AR: 1   | N: False  | Z: False
  DEBUG    root:control_unit.py:106 TICK: 751  | ld   | ARG: 36  | TYPE: default | PC: 47  | ACC: 1   | AR: 36  | N: False  | Z: False
  DEBUG    root:control_unit.py:106 TICK: 755  | dec  | ARG: -   | TYPE: none    | PC: 48  | ACC: 0   | AR: 47  | N: False  | Z: True
  DEBUG    root:control_unit.py:106 TICK: 762  | st   | ARG: 36  | TYPE: raw     | PC: 49  | ACC: 0   | AR: 36  | N: False  | Z: True
  DEBUG    root:control_unit.py:106 TICK: 766  | jz   | ARG: 51  | TYPE: default | PC: 51  | ACC: 0   | AR: 49  | N: False  | Z: True
  DEBUG    root:control_unit.py:106 TICK: 770  | hlt  | ARG: -   | TYPE: none    | PC: 52  | ACC: 0   | AR: 51  | N: False  | Z: True
